### ISSUE TITLE
pb with open_basedir on synology DSM 6

### ISSUE THREAD

**Issue by dauchinen** [rvolz/BicBucStriim#244]
Hello

I used BicBucStriim for 2 or 3 years but i needed to entirely reinstall my synology and BBS too.

At this times, i'va installed webstation, Apache 2,2, apache 2,4, PHP 5.6 and PHPmyadmin.

My books and my calibe metadata.db are on a folder called livres on my volume1 (= /volume1/livres)

When i test with http://mynas/bbs/installcheck.php

All the items are green except open_basedir (yellow) and calibre library check (red and NO to all 5 questions).

i've read the discussion https://github.com/rvolz/BicBucStriim/issues/226

and modify the open_basedir php parameters which are now : /tmp:/var/services/tmp:/var/services/web:/var/services/homes:/volume1/livres

But no change :-(
It seems that the modification of the open_base dir are note considered by bbs

i'm really disappointed because bbs was ok before the new installation.

**Comment by rvolz** [comment]
According to your description your problem might be more related to #237. I guess the access rights on the library directory (/volume1/livres) are too narrow. The HTTP user has no rights to read the library -- that's what the red light in the library check means. 

You have to give the webserver user (_http_) at least read/open access to the library directory and all its subdirectories and files.

**Comment by dauchinen** [comment]
that why i tought initially but (i'm just a windows user)

There is no hhtp user in the local user.
There is just a http in local group, which has permission to read/write on the shared directory.

On the page "installcheck" under the yellow point; yes open_base..... is exactly the same than the "standard" open_basedir of my synology, like if i hadn't change, like if my modifications are note saved.

**Comment by rvolz** [comment]
It happens that configuration changes are not applied immediately by the Synology. Maybe you should restart your NAS and see if the changed `open_basedir` paths appear in installcheck. Then the library check should work, too.

**Comment by armc** [comment]
I was having a similar problem with this.  The synology open_basedir variable does not appear to update properly... so bicbucstriim continuously shows a path that doesn't match the gui interface.

I was able to get it to work by creating a folder under homes and then ensuring that http group had access...

I couldn't find another way; however, if someone knows a solution, I would prefer to have my calibre library in a different share under /volume1/calibre.

Ideas?

**Comment by rvolz** [comment]
I just tried to replicate this. On my DSM 6.1 system ``open_basedir`` can be modified. That is with Apache 2.2 and PHP 5.6. 

However, if the problem persists you could try to change open_basedir with an editor: the configuration file seems to be /usr/syno/etc/packages/WebStation/php56/conf.d/user_settings.ini.

**Comment by armc** [comment]
Hello rvolz.  Thanks for the reply.  Done some more poking around here is what I found, but bottom line is I am still having the same issue with open_basedir.

>  Checked the user_settings.ini for php56.  While navigating there, I also saw a php70 directory.  I do not have php70 installed.  I noticed that bicbucstriim appears to be using that for the open_basedir setting, despite me not having it installed and WebStation set for php56.
>  I was able to confirm that when I made changes to the open_basedir through webstation, it was being written to the 5.6 settings file.
>  When I installed php70 and set that as the php... suddenly I was able to use the /volume1/Calibre/Library folder.... but I got quite a few more yellow gum balls on the installation check page...  The open_basedir on installationcheck did update to reflect the php70 setting...

I uninstalled everything (php70, 56, appache 22, 24, bbs (broke when installed appache 24)) and reinstalled apache 2.2, php5.6 and bicbucstriim.  Back to where I was...

Is there something in bicbucstriim that points toward the php70?

**Comment by rvolz** [comment]
Thx for the research! Interesting, no, there is no builtin preference for php70.

BicBucStriim uses the standard PHP function ``ini_get`` to get the value for the ``open_basedir`` configuration Apparently the values of the php70 package overwrite the php56 configuration? You could check what configuration files are loaded (and much more) in your system:

1. create a file called ``phpinfo.php`` in /volume1/web
1. enter the code  ``<?php phpinfo(); ?>``
1. call the file via browser

You'll get a full report of your PHP configuration with the loaded configuration files at the top. In my case -- I just updated to DSM 6.1 -- all configuration files are from the php56 directories, but maybe newly installed systems are different.

I guess that was the heads up to look into moving to PHP 7.

**Comment by armc** [comment]
Thanks for the tip....   it was _VERY_ helpful and key to my trouble shooting.

BLUF:  It appears as if my Synology DSM Webstation's php user_settings.ini is not being used by my php... which is referencing a /usr/local/etc/php56/conf.d/user_settings.ini.

Do you have any idea why there are so many different configuration locations for php?  I installed php through my synology package manager.  I am using the synocommunity repository for some things and also have installed entware-ng (but didn't install much from there).

I had been working with the user_settings.ini you referenced in the WebStation directory (
/usr/syno/etc/packages/WebStation/php56/conf.d/user_settings.ini); however, it appears that there is another location that php is defaulting to on my system.  /usr/local/etc/php56/conf.d/user_settings.ini

When comparing the two user_settings.ini... I can clearly see where the webstation ini is updated with what the GUI interface is set for.  The usr/local version did not reflect those updates.  

I have created a sim-link between the two so now changes through WebStation are reflected in my other location that appears to be the primary one referenced.  Hope this works... but still have to test.

`sudo ln -s /usr/syno/etc/packages/WebStation/php56/conf.d/user_settings.ini ./user_settings.ini`

**Comment by armc** [comment]
The linking of the webstation file to the /usr/local/etc/php56/conf.d/usr_settings.ini did the trick.  

Not sure if it is a good idea to do, but I noticed that if you used a custom open_basedir and left it blank... it turns off open_basedir...  

Still would like to know why I got a different php56 configuration directory...  but good to have a solution.

**Comment by iostrym** [comment]
hello, I'm a new cops user and I have exactly same problem than armc. open_basedir modification have no effect. I can put database in default value of open_basedir and it will work but when I want to put database in /valume2/ebook, it is no more working. as if open_basedir modifcaiton has no effect.
i'm not a linux expert but I found also strange that i have so many php configuration file........ 
so following your tips I modified 
/usr/local/etc/php56/conf.d/user_settings.ini

and after a reboot (before rebooting => no change) => it works !!! (about one weeks on this problem...)

if you know why synology config in webstation don't edit the correct file, I am very interested...
